src/app/clusters/ota-requestor/OTARequestor.h

```diff
@@ -76,15 +76,15 @@ class OTARequestor : public OTARequestorInterface, public BDXDownloader::StateDe
     // cleared, UpdateState is reset to Idle
     void CancelImageUpdate() override;
 
-    // Retrieve the default OTA provider list as an encoded list
-    CHIP_ERROR GetDefaultOtaProviderList(app::AttributeValueEncoder & encoder) override;
-
     // Clear all entries with the specified fabric index in the default OTA provider list
     CHIP_ERROR ClearDefaultOtaProviderList(FabricIndex fabricIndex) override;
 
     // Add a default OTA provider to the cached list
     CHIP_ERROR AddDefaultOtaProvider(
-        app::Clusters::OtaSoftwareUpdateRequestor::Structs::ProviderLocation::Type const & providerLocation) override;
+        const app::Clusters::OtaSoftwareUpdateRequestor::Structs::ProviderLocation::Type & providerLocation) override;
+
+    // Retrieve an iterator to the cached default OTA provider list
+    ProviderLocationList::Iterator GetDefaultOTAProviderListIterator(void) override { return mDefaultOtaProviderList.Begin(); }
```

Comment:
It would be helpful to have an example somewhere how this API should be used to get the set of provider locations.

Reply:
There is, take a look at the changes in this PR in ota-requestor-server.cpp